.github/workflows/test-on-pull-request.yml

```diff
@@ -0,0 +1,29 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Test React App On Pull Request
+
+on:
+  pull_request:
+    branches:
+      - 'feature/**'
+      - 'fix/**'
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm ci
+    - npm: install
```

Comment:
Is it necessary to run both these commands?

Reply:
No, it's not. the later command is more recommended for ci environments as it interacts with package.json file and doesn't alter package-log.json when the ci runs. I'll update the workflow to run the command: npm ci

Reply:
And I think I have an example. I'm getting an error when I run tests locally after merging: 

The react-scripts package provided by Create React App requires a dependency:
  "eslint": "^6.6.0"

The one found:
    D:\Monza\Projects\tictactoe\node_modules\eslint (version: 7.0.0)

Reply:
You'd need to downgrade your ESLint version